### ISSUE TITLE
Templating service files from ENV

### ISSUE THREAD

**Issue by esavier** [FedericoPonzi/Horust#48]
### Intro
Okay, as a user i would like to be able to use variables inside service.toml files.
From my understanding there is no possibility to do that out of the box, i.e.
setup i have depends on various things, i.e. where the user has its own repo, what's user's username is, etc. etc.

### In the field:
imagine following service file:
```
....
command = "../../repos/service/target/debug/service"
user = "esavier"
....

```
i would love to be able to modify it to look like that:
```
....
command = "${WORKSPACE}/target/${TARGET}/service"
user = "${USER}"
....

```
of course this is the variable format used by bash, and it's only here to provide an example and a background. 

###  What i would like to achieve?
flexibility, right now i have to go through some steps that will ensure that path exists (like symlinking, bleeh)

**Comment by FedericoPonzi** [comment]
Hey Mateusz! Thanks for this feature request :) 

This looks very interesting, my only concern is related to possible security issues that could come up with this. It's possible for an attacker to forge an ENV variable in order to run a different program then the initially intended one, and maybe run it as a root and thus allowing priviledge escalation.

For this reason, I would put it behind a feature flag like "--env" or something like that. WDYT?

Also, would you be interested to implement this feature? Again, thanks for opening this!

**Comment by esavier** [comment]
Hey again Federico!
Sorry for late response, what you said do make sense, so I sit down and thought it through for a bit.

About --env request first. 
Technically, the default way of running Horust would be from system side - as a root (lets say systemd replacement of sort). That's the default way, since all services may have `user = someone` config inside. That also means that only root should have access to the Horust's service directory, otherwise, --env will not help, since config can be crafted by anyone, do anything as anyone, and to start it you would have to just either restart the machine or restart Horust. 

The Other way around, assuming it is the user who starts the Horust, means Horust's uid is set to the user. If Services are runnable as a specific user - then great, that works, everyone's happy, but in case of `user = ` clause, it will fail to do so. Following that, ok, lets say program started, but someone injected some malicious code or path inside the env, that would mean that the user either have access to that path and code is runnable without permissions issue, or it will fail. If it works, that means the user had the whole time access to this path in level suitable to do whatever was done and the whole attack was unnecessary in the first place.

From the implementation's point of view (i.e. mechanics of templating itself), I think there is no need to fear permission problems since it would work basically as a very stupid version of `sed` and besides copypasting there would be no other processing going on. Of course, it depends on the library we choose but in the end it shouldn't do anything dangerous inside either.

TL;DR, technically linux access groups and permissions should solve all the issues. Of course depends on implementation, but I never saw escalation through ENV's that would work on a properly configured system.
But, but,

this is only speculation and should be tested on live example, since this may affect other features/enchantments.
I will try to find something suitable for this task and get back to you with working example. Of course, I believe --env should be added either way.

...and yeah i would love to work on that. Its kinda weird season to start r&d for me but I will do my best, and keep you posted about the updates :)

**Comment by FedericoPonzi** [comment]
Alright, sounds good to me! Feel free to open a PR, I'll happy to provide any needed help from there or from this issue whichever way you feel more useful.
(Forgot to hit publish on this comment)

**Comment by esavier** [comment]
Hey, PR is ready, went faster than expecte :)
I opted out from --env switch, since it would be weird to have a template file and not using the engine, also as i wrote before, there should not be any issues with permissions.
